react-ui/src/components/Layouts/InputWorkerTimes/CheckboxApplication.js

```diff
@@ -0,0 +1,135 @@
+  let firstDay = moment(dateContext).startOf('month').format('d') // day of week 0 ...1..2..6, 0 is sunday, 6 is saturday
+  return firstDay;
+}
+
+     
+  handleChange(event) {
+        var isChecked = event.target.checked;
+        var item = event.target.value;
+         
+        this.setState(prevState => ({ checkedDays: prevState.checkedDays.set(item, isChecked) }));
+  }
+     
+  handleSubmit(event) { // submits selected days and assign recurrenting dates based off boxes checked
+    /* console.log(this.state);  */ //test line
+   /*  const startDay = moment("9-5-2020");  */ 
+   const startDay = moment(document.getElementById("DATE").value)
+  /*  console.log(startDay.format("DD-MM-YYYY")); */ // Test line
+
+    let AvailableDates = [];
+    let currentDayNumber = moment(startDay).day();
+    /* console.log(currentDayNumber); */ //test line
+    let DaySelected = false; // variable for if checkboxes are checked or not
+
+    if(startDay.format("DD-MM-YYYY") === "Invalid date"){
+        alert("No date has been selected, please selected a day")
+        event.preventDefault();
+    }
+    else{
+    for(let i =1; i<= 7; i++){
+       let startofweek = moment(startDay).isoWeekday(0); // sets beginning of week to sunday
+       if(this.state.checkedDays.get(String(i)) === true){
+         DaySelected = true;
+        var AddDayCounter = i;
+
+        if(AddDayCounter<currentDayNumber){
+          AddDayCounter= AddDayCounter + 7; // to ensure that users are not adding recurring schedule to days before the selected date
+        }
+        
+        let StartingDay = startofweek.add(AddDayCounter,'day');
+        let NextWeek = "";
+
+
+        for(let k=0; k<=4; k++){
+          AvailableDates.push(StartingDay.format("DD-MM-YYYY")); // adjust formatting if needed
+          NextWeek = moment(StartingDay).add(7,"days");
+          StartingDay= NextWeek; 
+        }
+         console.log(AvailableDates);    // test line
+    } 
+  } 
+
+  if(DaySelected === false){
+    alert("No day has been checked, please selected a day");
+  }
+    event.preventDefault();  
+  }
+  }  
+
+  render() {
+   
+    return (
+      <div>
+          
+        <form className = {"Container"} onSubmit={this.handleSubmit}>
+        <h1>Recurring on</h1>
+        
+          { 
+            this.state.Days.map(item => (
+              <div className = {"checkboxes"}>
+                <label>
+                  <input className = "DayBox"
+                    type="checkbox"
+                    value={item.id}
+                    onChange={this.handleChange}
+                  /> {item.value}
+                </label>
+              </div>
+            ))
+          }
+           
+          <br/>
+          <input type = "date" id = "DATE"/>
+
+          <br/>
+          <input type="submit" value="Done" />
+        </form>
+         
+      </div>
+    );
+  }
+}
+   export default Check
```

Comment:
Hi Kyle,
Minor suggestion, right click and use Format on Visual Studio so that all of your code lines up.